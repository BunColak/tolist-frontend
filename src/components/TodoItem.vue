<template>
  <div>
    <label :class="todo.completed ? 'line-through' : ''">
      <input
        type="checkbox"
        v-model="todo.completed"
        @change="toggleTodo"
      />
      {{ todo.text }}
    </label>
  </div>
</template>

<script>
export default {
  name: 'TodoItem',
  props: {
    todo: Object
  },
  methods: {
    toggleTodo () {
      this.$emit('toggleTodo', this.todo.id)
    }
  }
}
</script>
