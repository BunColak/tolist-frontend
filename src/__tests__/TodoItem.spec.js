import { shallowMount } from '@vue/test-utils'
import TodoItem from '@/components/TodoItem'

const mockTodo = {
  id: 1,
  text: 'Test Text',
  completed: false
}

describe('TodoItem', () => {
  it('should render without problems', () => {
    const wrapper = shallowMount(TodoItem, { propsData: { todo: mockTodo } })
    expect(wrapper).toMatchSnapshot()
  })

  it('should emit toggle with correct parameters', async () => {
    const wrapper = shallowMount(TodoItem, { propsData: { todo: mockTodo } })
    wrapper.find('input').trigger('click')
    await wrapper.vm.$nextTick()
    expect(wrapper.emitted().toggleTodo).toBeTruthy()
    expect(wrapper.emitted().toggleTodo.length).toBe(1)
    expect(wrapper.emitted().toggleTodo[0]).toEqual([mockTodo.id])
  })
})
