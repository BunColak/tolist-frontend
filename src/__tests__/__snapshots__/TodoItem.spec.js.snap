// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem should render without problems 1`] = `
<div><label class=""><input type="checkbox">
    Test Text
  </label></div>
`;
